src/main/java/org/checkerframework/specimin/SpeciminRunner.java

```diff
@@ -236,7 +237,10 @@ private static void performMinimizationImpl(
     Map<String, String> nonPrimaryClassesToPrimaryClass = new HashMap<>();
     SourceRoot sourceRoot = new SourceRoot(Path.of(root));
     sourceRoot.tryToParse();
-    for (CompilationUnit compilationUnit : sourceRoot.getCompilationUnits()) {
+    //getCompilationUnits does not seem to include all files, causing some to be deleted
+    for (ParseResult<CompilationUnit> res : sourceRoot.getCache()) {
+      CompilationUnit compilationUnit =
+          res.getResult().orElseThrow(() -> new RuntimeException("" + res.getProblems()));
```

Comment:
Why `"" + ...` rather than just a call to `toString()`?

Reply:
done